### DISCUSSION TITLE
Possibile to set row height based on text content of cell?

### DISCUSSION THREAD

**Discussion by mdodge-ecgrow** [exceljs/exceljs#2775] [Q&A]
I am creating a spreadsheet with this library.  I have one cell with a large chunk of text, looks like 552 characters.  I have text wrap on that cell, but after the first line of text, the rest is not visible.  At first I was looking for a way to wrap the text onto multiple rows, but that looks like a much harder problem to solve.  So now I'm looking for a way to set the height of the row based on that one cell?

I already found an answer here to auto-calculate the column width based on the largest text in that column: https://github.com/exceljs/exceljs/discussions/2535#discussioncomment-7075656
Hoping for something similar with row height.

**Comment by tilamit** [comment]
Did you get any solution to this or anyway managed it?

**Reply by mdodge-ecgrow** [reply]
I ended up writing some custom code to calculate the height of that row based on the length of the text.  I used some hardcoded numbers and it is certainly not perfect, but it gets the job done.  The code may not mean much to your or help you much, but I'll paste it here anyways.
`
let rowHeight = 15;
const maxLineChars = 75;
if (lineCount === 1) {
	rowHeight =
		noteLength > maxLineChars
			? Big(Big(noteLength).div(Big(maxLineChars)).toFixed(0, 3))
				.times(Big(15))
				.toNumber()
			: 15;
} else {
	if (noteLength < maxLineChars) {
		rowHeight = 15 * lineCount;
	} else {
		const divider = lineCount > 10 ? 2.2 : 2.4;
		rowHeight = Math.max(
			15,
			Number(Big(notes.length).div(divider).toFixed(0, 3))
		);
	}
}
`
